### PR TITLE
Use TestPurpose in LuckyPeople

### LINKED ISSUE #173: Use TestPurpose in LuckyPeople
- [x] Replace old style purpose models with proper Test Purposes
- [x] Add a test using IncrementRows Purpose to test for luck based on previous people's genders (MMMMMF or FFFFFM) (#157)
- [x] Update documentation - [Modelling Example Lucky People (Solution)](https://github.com/TorXakis/TorXakis/wiki/Modelling-Example-Lucky-People-(Solution))

### PR DESCRIPTION
**Pull request by keremispirli**
Closes #173

### CODE REVIEW

File: test/examps/LuckyPeople/TestLuckyPeople.bat [hunk 1]
```diff
@@ -3,26 +3,38 @@
...
+echo ------- End LuckyPeople Random Test
+
+echo ------- Start LuckyPeople By Gender Test
+start /min java -cp %ASSIGN% LuckyPeople
+call %TORXAKISPORT% %PORT% %SOL%\LuckyPeople.txs < %TEST%\LuckyPeopleByGender_Purpose.txscmd
+TIMEOUT /T 4 /NOBREAK
+move /Y .txs.%PORT%.err.log ByGender.txs.%PORT%.err.log
+echo ------- End LuckyPeople By Gender Test
+
+REM echo ------- Start LuckyPeople Overflow Test
```
Comment (capitanbatata) [inline test/examps/LuckyPeople/TestLuckyPeople.bat]: Why are we commenting this out? Can't we just remove this?

Comment (keremispirli) [reply test/examps/LuckyPeople/TestLuckyPeople.bat]: I'm not sure about this Overflow test; it requires another 300 steps to trigger byte overflow for both genders (128 per gender, round up to make up for Quiescence's) and current test purpose is not enough to test it - it should ensure that when switched to other gender the new person is not lucky by some other criteria.

I am going to open an issue for this, so I wanted to keep it as reference. We can remove it after we decide what to do about this case.

Comment (keremispirli) [reply test/examps/LuckyPeople/TestLuckyPeople.bat]: Thinking again, I will just remove it. These lines don't add value even as reference.

**Comment by keremispirli** [comment]
@pjljvandelaar Could you review *PurposeByteOverflow*? Is it correct? And is there a way to test both Male and Female overflow, other than using separate Test Purposes?

@capitanbatata and @pjljvandelaar:
I didn't want to add Overflow test to sqatt for two reasons:

1. It requires some ~700 steps to trigger byte overflow for both genders (128 cases = 256 steps per gender, round up to make up for Quiescence's). This will increase test time. Do we want it anyway?

2. Current test purpose is not enough to test it - it should ensure that when switched to other gender the new person is not lucky by some other criteria. If we use LuckyPeopleOverflow as SUT and expect failure (which I think makes more sense) then this has the potential to cause occasional false alarms. 

Should we:
 - write a more complex _byteOverflow_ process?
 - just use LuckyPeople as SUT?
 - not write test for this case at all?

**Comment by capitanbatata** [comment]
I would prefer to avoid making the integration tests slower. If we want to run longer tests we could think about setting up a Linux box at the office, and run those there...

**Comment by pjljvandelaar** [comment]
Not all examples must be tested in the development acceptance tests.
Of course, the examples must be regularly tested.... but other alternatives for testing them is possible!

**Comment by keremispirli** [comment]
That pesky SpecProduce keeps failing. I'm almost done with getting rid of it though (#227).

**Pull request merged**